tests/test_pydantic_factory.py

```diff
@@ -717,18 +717,49 @@ class C(BaseModel):
     assert CFactory.build()
 
 
-def test_constrained_union_types() -> None:
+@pytest.mark.skipif(IS_PYDANTIC_V2, reason="pydantic 1 only test")
+def test_constrained_union_types_pydantic_v1() -> None:
     class A(BaseModel):
         a: Union[Annotated[List[str], MinLen(100)], Annotated[int, Ge(1000)]]
         b: Union[List[Annotated[str, MinLen(100)]], int]
         c: Union[Annotated[List[int], MinLen(100)], None]
-        d: Union[Annotated[List[int], MinLen(100)], Annotated[List[str], MinLen(100)]]
-        e: Optional[Union[Annotated[List[int], MinLen(10)], Annotated[List[str], MinLen(10)]]]
+        d: Union[Annotated[List[int], MinLen(100)], Annotated[List[str], MaxLen(99)]]
+        e: Optional[Union[Annotated[List[int], MinLen(10)], Annotated[List[str], MaxLen(9)]]]
         f: Optional[Union[Annotated[List[int], MinLen(10)], List[str]]]
+        g: Optional[
+            Union[
+                Annotated[List[int], MinLen(10)],
+                Union[Annotated[List[str], MaxLen(9)], Annotated[Decimal, Field(max_digits=4, decimal_places=2)]],
+            ]
+        ]
+
+    AFactory = ModelFactory.create_factory(A, __allow_none_optionals__=False)
+
+    assert AFactory.build()
```

Comment:
My fix only seems to work for pydantic 2, adding a call to `coverage` here crashes 
```suggestion
    assert AFactory.build()
    assert list(AFactory.coverage())
```
(Note that this also fails on `main`)